Resources/Prototypes/Entities/Objects/Weapons/Guns/Snipers/snipers.yml

```diff
@@ -69,36 +69,58 @@
   id: Musket
   description: This should've been in a museum long before you were born. Uses .60 anti-materiel ammo.
   components:
+  - type: Sharp
   - type: Item
     size: Large
   - type: Sprite
     sprite: Objects/Weapons/Guns/Snipers/musket.rsi
+    state: base
   - type: Clothing
     sprite: Objects/Weapons/Guns/Snipers/musket.rsi
   - type: Gun
+    useKey: false
     selectedMode: SemiAuto
     availableModes:
     - SemiAuto
+  - type: UseDelayOnShoot
+  - type: UseDelay
+    delay: 8 #it's a musket
   - type: BallisticAmmoProvider
     whitelist:
       tags:
       - CartridgeAntiMateriel
     capacity: 1
     proto: CartridgeAntiMateriel
-
+  - type: MeleeWeapon
```

Comment:
Does this work? I thought the reason it wasn't on the mosin was that having a meleeweapon and gun component caused issues.

Reply:
I like the flintlock sprite, its a big improvement.

Reply:
it works, I just had to make the musket fire on right click.